### ISSUE TITLE
Make ArrayOps not extend IndexedSeqOps

### ISSUE THREAD

**Issue by julienrf** [scala/collection-strawman#464]

**Comment by lrytz** [comment]
So the goal here would be to get rid of unwanted inherited methods, such as the overload of map that works without a ClassTag. Example:

```
scala> val a = Array("a", "b").map(_.toUpperCase)
a: Array[String] = Array(A, B)

scala> val b = Array("a", "b").map(_.toUpperCase) : Iterable[String]
b: Iterable[String] = Vector(A, B)

scala> a : Iterable[String]
res16: Iterable[String] = WrappedArray(A, B)
```

But if ArrayOps no longer extends IterableOps, all the inherited operations that would work fine have to be re-implemented. There's no parent trait that would enforce the same interface - a problem that we already have between Iterator and IterableOps (https://github.com/scala/collection-strawman/issues/465).

@julienrf you looked at having parent traits that would group operations by their kind (reducing, transforming, self-transforming)? Is it out of the question?

**Comment by julienrf** [comment]
I think we could probably have a parent trait for reducing operations, because unlike operations that return `C` or `CC[?]`, there is no need for taking a type parameter (and thus no need to refine it at each level of the hierarchy).

**Comment by julienrf** [comment]
Closed by #479.
